crates/nu-cli/src/completions/custom_completions.rs

```diff
@@ -102,17 +103,22 @@ impl<T: Completer> Completer for CustomCompletion<T> {
                         {
                             completion_options.case_sensitive = case_sensitive;
                         }
-                        if let Some(positional) =
-                            options.get("positional").and_then(|val| val.as_bool().ok())
-                        {
-                            completion_options.positional = positional;
-                        }
                         if let Some(algorithm) = options
                             .get("completion_algorithm")
                             .and_then(|option| option.coerce_string().ok())
                             .and_then(|option| option.try_into().ok())
                         {
                             completion_options.match_algorithm = algorithm;
+                            if let Some(positional) =
+                                options.get("positional").and_then(|val| val.as_bool().ok())
+                            {
+                                log::warn!("Use of the positional option is deprecated. Use the substring match algorithm instead.");
+                                if !positional
+                                    && completion_options.match_algorithm == MatchAlgorithm::Prefix
+                                {
+                                    completion_options.match_algorithm = MatchAlgorithm::Substring
+                                }
+                            }
```

Comment:
Sorry, my earlier suggestion was wrong, because this warning will only be given if you specify `positional` *and* the completion algorithm is specified. Better to define `positional` outside the `if let Some(algorithm)` block (and also log the warning there). And then here, you'd just set the algorithm to substring, no logging:
```suggestion
                            if !positional
                                    && completion_options.match_algorithm == MatchAlgorithm::Prefix
                            {
                                completion_options.match_algorithm = MatchAlgorithm::Substring
                            }
                        }
```